pkg/api/types.go

```diff
@@ -1889,6 +1889,8 @@ type EndpointAddress struct {
 	// Optional: Hostname of this endpoint
 	// Meant to be used by DNS servers etc.
 	Hostname string `json:"hostname,omitempty"`
+	// Optional: Node hosting this endpoint
+	NodeName string `json:"nodeName,omitempty"`
```

Comment:
yeah I didn't realize. I used to do it for fields that can be sensibly defaulted, like unset rc.replicas defaults to 1 because we detect a nil pointer vs a pointer to 0.